### PR TITLE
Group export: Handle empty groups correctly

### PR DESCRIPTION
**Pull request by namadabu**
## Description
When exporting a group that has no members we end up exporting all patient data. This PR updates the code to handle this scenario correctly and export nothing.

## Related issues
Addresses #1634 

## Testing
Manual testing on local deployment. Added E2E test.

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch

### CODE REVIEW

**Review by LTA-Thinking — APPROVED** [review]

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ExportDataValidationTests.cs [hunk 1]
```diff
@@ -529,5 +544,17 @@ private async Task<(Dictionary<(string resourceType, string resourceId), Resourc
 
             return (resourceDictionary, groupId);
         }
+
+        private async Task<string> CreateGroupWithoutPatientIds()
```
Comment (Ivanidzo4ka) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ExportDataValidationTests.cs:548]: >CreateGroupWithoutPatientIds [](start = 35, length = 28)

super nit: I would probably make it local function.

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs [hunk 2]
```diff
@@ -677,7 +677,19 @@ private async Task<SearchResult> GetGroupPatients(string groupId, List<Tuple<str
...
+                HashSet<string> patientIds = await _groupMemberExtractor.GetGroupPatientIds(groupId, groupMembershipTime, cancellationToken);
+
+                if (patientIds.Count == 0)
+                {
+                    _logger.LogInformation($"Group {groupId} does not have any patient ids as members.");
+                    SearchResult emptySearchResult = new SearchResult(
+                        results: new List<SearchResultEntry>(),
+                        continuationToken: null,
+                        sortOrder: null,
+                        unsupportedSearchParameters: new List<Tuple<string, string>>());
```
Comment (Ivanidzo4ka) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs:689]: nit: why not: `new SearchResult(0, new List<Tuple<string, string>>())`?

**Review by Ivanidzo4ka — APPROVED** [review]
:shipit:

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs (hunk 2 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs:689]: Depends on how you feel, I would even step further and create
```
public static SearchResult Empty(IReadOnlyList<Tuple<string, string>> unsupportedSearchParameters=null)=>
new SearchResult(0, unsupportedSearchParameters?? new List<Tuple<string,string>>()
```
method inside search result class and use it among other places in code.  (I can see few cases where we create empty search result other than this part of code like `CreateEmptySearchResponse` method  or tests in SearchServiceTests.)

Comment (namadabu) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs]: Added an extension and updated code to use the new `SearchResult.Empty()` method.

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ExportDataValidationTests.cs (hunk 1 above)
Comment (namadabu) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ExportDataValidationTests.cs]: Updated to make it a local function.

**Pull request merged**